rules/do-you-catch-and-re-throw-exceptions-properly/rule.md

```diff
@@ -1,68 +1,73 @@
-redirects: []
-created: 2013-09-11T21:16:44.000Z
-archivedreason: null
-guid: d8992617-dcae-4dca-b775-9e417507d1b5
----
-
-A good catch and re-throw will make life easier while debugging, a bad catch and re-throw will ruin the exception's stack trace and make debugging difficult.
-
-Catch and rethrow where you can usefully add more information that would save a developer having to work through all the layers to understand the problem.
-
-<!--endintro-->
-
-```cs
-catch {} 
-
-catch (SomeException) {} 
-
-catch { throw; } 
-
-catch (SomeException) { throw; } 
-```
-::: bad
-Bad Example - Never use an empty catch block. Do something in the block or remove it.
-:::
-
-```cs
-catch (SomeException ex) { throw ex; } 
-
-catch (SomeException ex) { someMethod(); throw ex; } 
-```
-::: bad
-Bad Example - Never re-throw exceptions by passing the original exception object. Wrap the exception or use throw; instead.
-:::
-
-```cs
-catch (SomeException) 
-{ 
-     someMethod(); 
-     throw; 
-}
-```
-::: good
-Good Example - Calling throw  
-:::
-
-If you are following the [Clean Architecture](https://github.com/jasontaylordev/CleanArchitecture) pattern - catching and rethrowing is useful for preventing your Infrastructure details from leaking into your Application e.g. we use a SQL server
-
-```cs
+---
+type: rule
+title: Do you catch and re-throw exceptions properly?
+uri: do-you-catch-and-re-throw-exceptions-properly
+authors:
+  - title: Adam Cogan
+    url: https://ssw.com.au/people/adam-cogan
+  - title: Matt Wicks
+    url: https://www.ssw.com.au/people/matt-wicks
+  - title: Piers Sinclair
+    url: https://www.ssw.com.au/people/piers-sinclair
+related: []
+redirects: []
+created: 2013-09-11T21:16:44.000Z
+archivedreason: null
+guid: d8992617-dcae-4dca-b775-9e417507d1b5
+---
+
+A good catch and re-throw will make life easier while debugging, a bad catch and re-throw will ruin the exception's stack trace and make debugging difficult.
+
+Catch and rethrow where you can usefully add more information that would save a developer having to work through all the layers to understand the problem.
+
+<!--endintro-->
+
+```cs
+catch {} 
+
+catch (SomeException) {} 
+
+catch { throw; } 
+
+catch (SomeException) { throw; } 
+```
+::: bad
+Bad Example - Never use an empty catch block. Do something in the block or remove it.
+:::
+
+```cs
+catch (SomeException ex) { throw ex; } 
+
+catch (SomeException ex) { someMethod(); throw ex; } 
+```
+::: bad
+Bad Example - Never re-throw exceptions by passing the original exception object. Wrap the exception or use throw.
+:::
+
+The "Throw(ex)" method resets the stack trace, causing the error to appear as if it originated from the line where "Throw(ex)" is used.
+
+```cs
+catch (SomeException) 
+{ 
+     someMethod(); 
+     throw; 
+}
+```
+
+The "Throw" method throws the current exception without altering the stack trace, allowing you to preserve information about the original exception, which is the desired behavior.

```

Comment:
I think this part is unnecessary since the sentence above is already enough

Reply:
Done removed